app/src/androidTestKiwix/java/org/kiwix/kiwixmobile/tests/DownloadTest.java

```diff
@@ -38,104 +40,97 @@
 
   @BeforeClass
   public static void beforeClass() {
-    IdlingPolicies.setMasterPolicyTimeout(350, TimeUnit.SECONDS);
-    IdlingPolicies.setIdlingResourceTimeout(350, TimeUnit.SECONDS);
+    IdlingPolicies.setMasterPolicyTimeout(180, TimeUnit.SECONDS);
+    IdlingPolicies.setIdlingResourceTimeout(180, TimeUnit.SECONDS);
     Espresso.registerIdlingResources(KiwixIdlingResource.getInstance());
   }
 
   @Before
   public void setUp() {
-
   }
 
   @Test
-  public void downloadTest() {
-    ViewInteraction appCompatTextView = onView(
-        allOf(withText("Device"), isDisplayed()));
-    appCompatTextView.perform(click());
+  public void downloadTest() throws Throwable {
+    enterHelp();
+    clickOn(R.string.menu_zim_manager);
 
-    try {
-      onData(withContent("ray_charles")).inAdapterView(withId(R.id.zimfilelist)).perform(longClick());
-      onView(withId(android.R.id.button1)).perform(click());
-    } catch (RuntimeException e) {
+    clickOn(R.string.local_zims);
 
-    }
+    allowPermissionsIfNeeded();
+
+    Activity currentActivity = getCurrentActivity(mActivityTestRule);
 
-    ViewInteraction appCompatTextView2 = onView(
-        allOf(withText("Online"), isDisplayed()));
-    appCompatTextView2.perform(click());
+    deleteZimIfExists("ray_charles", R.id.zimfilelist);
+
+    clickOn(R.string.remote_zims);
 
     try {
-      onView(withId(R.id.network_permission_button)).perform(click());
-      Log.d("kiwixDownloadTest", "Clicked Network Permission Button");
+      clickOn(R.id.network_permission_button);
     } catch (RuntimeException e) {
-      Log.d("kiwixDownloadTest", "Failed to click Network Permission Button", e);
+      Log.d(KIWIX_DOWNLOAD_TEST, "Failed to click Network Permission Button", e);
     }
 
-    Spoon.screenshot(mActivityTestRule.getActivity(), "Before-checking-for-ZimManager-Main-Activity");
+    Spoon.screenshot(currentActivity, "Before-checking-for-ZimManager-Main-Activity");
     ViewInteraction viewPager2 = onView(
-        allOf(withId(R.id.container),
-            withParent(allOf(withId(R.id.zim_manager_main_activity),
-                withParent(withId(android.R.id.content)))),
-            isDisplayed()));
-    Spoon.screenshot(mActivityTestRule.getActivity(), "After-the-check-completed");
-
-    try {
-      Thread.sleep(20000);
-    } catch (InterruptedException e) {
-      e.printStackTrace();
-    }
+            allOf(withId(R.id.container),
+                    withParent(allOf(withId(R.id.zim_manager_main_activity),
+                            withParent(withId(android.R.id.content)))),
+                    isDisplayed()));
+    Spoon.screenshot(currentActivity, "After-the-check-completed");
 
     onData(withContent("ray_charles")).inAdapterView(withId(R.id.library_list)).perform(click());
 
     try {
-      onView(withId(android.R.id.button1)).perform(click());
+      clickDialogPositiveButton();
     } catch (RuntimeException e) {
+      // Confirmation not shown
```

Comment:
As you know, swallowing exceptions quietly and continuing can be risky (as the state of the system may not match that which the test expects) and hard to diagnose as nothing's mentioned or reported. I prefer to log these instances when they occur so we have the opportunity to review logs afterwards if we're trying to diagnose problems, etc.